darts/models/forecasting/regression_ensemble_model.py

```diff
@@ -170,17 +244,26 @@ def fit(
         )
 
         # prepare the forecasting models for further predicting by fitting them with the entire data
+        if self.retrain_forecasting_models:
+            # Some models may need to be 'reset' to allow being retrained from scratch, especially torch-based models
+            self.forecasting_models: List[ForecastingModel] = [
+                model.untrained_model() for model in self.forecasting_models
+            ]
 
-        # Some models (incl. Neural-Network based models) may need to be 'reset' to allow being retrained from scratch
-        self.models = [model.untrained_model() for model in self.models]
-
-        for model in self.models:
-            kwargs = dict(series=series)
-            if model.supports_past_covariates:
-                kwargs["past_covariates"] = past_covariates
-            if model.supports_future_covariates:
-                kwargs["future_covariates"] = future_covariates
-            model.fit(**kwargs)
+            for model in self.forecasting_models:
+                model._fit_wrapper(
+                    series=series,
+                    past_covariates=past_covariates
+                    if model.supports_past_covariates
+                    else None,
+                    future_covariates=future_covariates
+                    if model.supports_future_covariates
+                    else None,
+                )
+        # update training_series attribute to make predict() behave as expected
+        else:
+            for model in self.forecasting_models:
+                model.training_series = series if is_single_series else None
```

Comment:
I would rather pass the series at prediction time (with `series=`) than mutating the models.
As far as I can tell, EnsembleModel.predict() passes the series to the forecasting `model.predict()`.
Since retrain_forecasting_models=False, we have only global models here which allow passing a series at prediction time.

Reply:
If the global models were (previously) fitted using single series (hence saved, with a time index potentially different than the one being fitted) and later, `predict()` is called with `series=None`, then the forecasting models could produce covariates with different time indexes and the ensemble won't work. I remember having this issue when running some tests

To prevent this, we could eventually retrieve the `ensemble.regression_model.training_series` when `series=None` instead of each `model.training_series` when `_make_multiple_predictions()` is called in `ensemble.predict()`.